config.md

```diff
@@ -25,19 +25,61 @@ Changing it means creating a new derived image, instead of changing the existing
 
 ### Layer DiffID
 
-A layer DiffID is a SHA256 digest over the layer's uncompressed tar archive and serialized in the descriptor digest format, e.g., `sha256:a9561eb1b190625c9adb5a9513e72c4dedafc1cb2d4c5236c9a6957ec7dfd5a9`.
+A layer DiffID is the digest over the layer's uncompressed tar archive and serialized in the descriptor digest format, e.g., `sha256:a9561eb1b190625c9adb5a9513e72c4dedafc1cb2d4c5236c9a6957ec7dfd5a9`.
 Layers must be packed and unpacked reproducibly to avoid changing the layer DiffID, for example by using tar-split to save the tar headers.
 
 NOTE: Do not confuse DiffIDs with [layer digests](manifest.md#image-manifest-property-descriptions), often referenced in the manifest, which are digests over compressed or uncompressed content.
 
 ### Layer ChainID
 
 For convenience, it is sometimes useful to refer to a stack of layers with a single identifier.
-This is called a `ChainID`.
-For a single layer (or the layer at the bottom of a stack), the
-`ChainID` is equal to the layer's `DiffID`.
-Otherwise the `ChainID` is given by the formula:
-`ChainID(layerN) = SHA256hex(ChainID(layerN-1) + " " + DiffID(layerN))`.
+While a layer's `DiffID` identifies a single changeset, the `ChainID` identifies the subsequent application of those changesets.
+This ensures that we have handles referring to both the layer itself, as well as the result of the application of a series of changesets.
+Use in combination with `rootfs.diff_ids` while applying layers to a root filesystem to uniquely and safely identify the result.
+
+#### Definition
+
+The `ChainID` of an applied set of layers is defined with the following recursion:
+
+```
+ChainID(L₀) =  DiffID(L₀)
+ChainID(L₀|...|Lₙ₋₁|Lₙ) = Digest(ChainID(L₀|...|Lₙ₋₁) + " " + DiffID(Lₙ))
+```
+
+For this, we define the binary `|` operation to be the result of applying the right operand to the left operand.
+For example, given base layer `A` and a changeset `B`, we refer to the result of applying `B` to `A` as `A|B`.
```

Comment:
> > You can avoid defining | and just define ChainID to act on an array of strings.
>
> But that is not the definition of a ChainID.

Not as your PR stands with 0fa3cc5, but with a string-based ChainID definition, it's easy enough to [couple back in your choice of `DiffID(…)` implementation][1]:

    ChainID([DiffID(L₀), DiffID(L₁), …, DiffID(Lₙ)])

On the other hand, it's clearly impossible to *decouple* the `ChainID` from layers if you require the arguments to be layers ;).

Your implementation seems to agree by [accepting `Digest`s][2] and then [treating them as opaque strings][3].  There is nothing layer-specific in the current `identity/chainid.go` code, and I like that, and think the Markdown definition should follow that example.

[1]: https://github.com/opencontainers/image-spec/pull/586#discussion_r103806487
[2]: https://github.com/opencontainers/image-spec/blob/v1.0.0-rc5/identity/chainid.go#L30
[3]: https://github.com/opencontainers/image-spec/blob/v1.0.0-rc5/identity/chainid.go#L57-L61